tests/test_other.py

```diff
@@ -30,15 +30,18 @@ def __exit__(self, type, value, traceback):
       try_delete(self.directory)
 
 class clean_write_access_to_canonical_temp_dir(object):
+  def __init__(self, dir=CANONICAL_TEMP_DIR):
+    self.canonical_temp_dir = dir
+
   def clean_emcc_files_in_temp_dir(self):
-    for x in os.listdir(CANONICAL_TEMP_DIR):
+    for x in os.listdir(self.canonical_temp_dir):
```

Comment:
`self.canonical_temp_dir` is defined as a function in `runner.py`, but used as a string here?

Reply:
Ah, the class `clean_write_access_to_canonical_temp_dir` has its own field `canonical_temp_dir` which is a string.